src/main/java/be/syntra/devshop/DevshopFront/services/CartServiceImpl.java

```diff
@@ -122,12 +147,33 @@ private void setCartToFinalized(CartDto cartDto) {
 
     @Override
     public BigDecimal getCartTotalPrice(CartDto currentCart) {
-        return currentCart.getProducts().stream()
-                .map(product -> {
-                    BigDecimal pricePerProduct = product.getPrice();
-                    int totalInCart = product.getTotalInCart();
-                    return pricePerProduct.multiply(new BigDecimal(totalInCart));
+        return currentCart.getCartProductDtoList().stream()
+                .map(cartProductDto -> {
+                    BigDecimal productPrice = cartProductDto.getProductDto().getPrice();
```

Comment:
productPrice could be inlined

Reply:
inlined getCount() also and then followed intellij's advice to prettify it, i think its fairly more readable now